tests/integration/nm/ovs_test.py

```diff
@@ -0,0 +1,185 @@
+            OB.PORT_SUBTREE: []
+        }
+    }
+
+
+@pytest.fixture
+def bridge_default_config():
+    return {
+        OB.CONFIG_SUBTREE: {
+            OB.OPTIONS_SUBTREE: {
+                OB.FAIL_MODE: '',
+                OB.MCAST_SNOOPING_ENABLED: False,
+                OB.RSTP: False,
+                OB.STP: False
+            },
+            OB.PORT_SUBTREE: []
+        }
+    }
+
+
+def test_create_and_remove_minimum_config_bridge(bridge_minimum_config,
+                                                 bridge_default_config):
+    bridge_desired_state = bridge_minimum_config
+
+    with _bridge_interface(bridge_desired_state):
+        bridge_current_state = _get_bridge_current_state()
+        assert bridge_current_state == bridge_default_config
+        assert not bridge_current_state[OB.CONFIG_SUBTREE][OB.PORT_SUBTREE]
+
+    assert not _get_bridge_current_state()
+
+
+def test_create_and_remove_bridge(eth1_up, bridge_default_config):
+    bridge_desired_state = bridge_default_config
+
+    eth1_port = {
+        OB.PORT_NAME: 'eth1',
+        OB.PORT_TYPE: 'system',
+        # OVS vlan/s are not yet supported.
+        # OB.PORT_VLAN_MODE: None,
+        # OB.PORT_ACCESS_TAG: 0,
+    }
+
+    bridge_desired_state[OB.CONFIG_SUBTREE][OB.PORT_SUBTREE].append(eth1_port)
+
+    with _bridge_interface(bridge_desired_state):
+        bridge_current_state = _get_bridge_current_state()
+        assert bridge_desired_state == bridge_current_state
+
+    assert not _get_bridge_current_state()
+
+
+@contextmanager
+def _bridge_interface(state):
+    try:
+        _create_bridge(state)
+        yield
+    finally:
+        _delete_iface(BRIDGE0)
+        for p in state[OB.CONFIG_SUBTREE][OB.PORT_SUBTREE]:
+            _delete_iface(nm.ovs.PORT_PROFILE_PREFIX + p[OB.PORT_NAME])
+            _delete_iface(p[OB.PORT_NAME])
+
+
+def _get_bridge_current_state():
+    nm.nmclient.client(refresh=True)
+    nmdev = nm.device.get_device_by_name(BRIDGE0)
+    devices_info = [(dev, nm.device.get_device_common_info(dev))
+                    for dev in nm.device.list_devices()]
+    return nm.ovs.get_bridge_info(nmdev, devices_info) if nmdev else {}
+
+
+def _create_bridge(bridge_desired_state):
+    br_options = nm.ovs.translate_bridge_options(bridge_desired_state)
+    iface_bridge_settings = _get_iface_bridge_settings(br_options)
+
+    with mainloop():
+        _create_bridge_iface(iface_bridge_settings)
+        ports_state = bridge_desired_state[OB.CONFIG_SUBTREE][OB.PORT_SUBTREE]
+        for port_state in ports_state:
+            _attach_port_to_bridge(port_state)
+
+
+def _attach_port_to_bridge(port_state):
+    port_profile_name = nm.ovs.PORT_PROFILE_PREFIX + port_state['name']
+
+    _create_proxy_port(port_profile_name, port_state)
+    _enslave_interface(port_profile_name, port_state)
+
+
+def _enslave_interface(port_profile_name, port_state):
+    slave_nmdev = nm.device.get_device_by_name(port_state['name'])
```

Comment:
Use `OB.PORT_NAME`

Reply:
`port_nmdev`...

Reply:
In this context (and detail), `port` does not fit. We have both an interface and a port, ovs/nm differentiating between them.

Changed it to `iface_nmdev`.